app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js

```diff
@@ -1026,19 +1028,22 @@ define([
                 $widget._EnableProductMediaLoader($this);
                 mediaSuccessCallback($widget.options.mediaCache[mediaCacheKey]);
             } else {
-                mediaCallData.isAjax = true;
-                $widget._XhrKiller();
-                $widget._EnableProductMediaLoader($this);
-                $widget.xhr = $.ajax({
-                    url: $widget.options.mediaCallback,
-                    cache: true,
-                    type: 'GET',
-                    dataType: 'json',
-                    data: mediaCallData,
-                    success: mediaSuccessCallback
-                }).done(function () {
+                var checkAdditionalData = this.options.jsonSwatchConfig.additional_data[currentAttributeId];
+                if (checkAdditionalData.update_product_preview_image == 1) {
```

Comment:
I think it's better to move this check to `_onClick` and `_onChange` (to handle select as well) functions and do not call `_loadMedia` at all in case product image update is set to No. (as there are two ways to load media - ajax and json config).
Also, with this approach, the `currentAttributeId` global variable is not needed.

Reply:
Hello @sivaschenko 
I have update app/code/Magento/Swatches/view/frontend/web/js/swatch-renderer.js 
move check into _OnClick